interfaces/apparmor/spec.go

```diff
@@ -21,22 +21,25 @@ package apparmor
 
 import (
 	"github.com/snapcore/snapd/interfaces"
+
+	"sort"
+	"strings"
 )
 
 // Specification assists in collecting apparmor entries associated with an interface.
 type Specification struct {
 	// snippets are indexed by security tag.
-	snippets     map[string][][]byte
+	snippets     map[string][]string
 	securityTags []string
 }
 
 // AddSnippet adds a new apparmor snippet.
-func (spec *Specification) AddSnippet(snippet []byte) error {
+func (spec *Specification) AddSnippet(snippet string) error {
```

Comment:
Why do we return an error here?  This means we need tons of error checks that simply cannot ever happen.

I think we can drop this and cleanup the whole logic accordingly.